Sources/Kommunicate/Classes/CustomPreChatFormViewController.swift

```diff
@@ -258,6 +309,74 @@ open class CustomPreChatFormViewController: UIViewController {
         }
     }
     
+    
+    var transparentView = UIView()
+    var tableView = UITableView()
+    var selectedButton = UIButton()
+    var selectedDataSource = [String]()
+    private func addTransparent(_ rect: CGRect){
+        
+        let window = UIApplication.shared.keyWindow
+        transparentView.frame = window?.frame ?? self.view.frame
+        self.view.addSubview(transparentView)
+        
+        tableView.frame = CGRect(x: rect.origin.x, y: rect.origin.y + rect.height, width: rect.width, height: 0)
+        self.view.addSubview(tableView)
+        tableView.layer.cornerRadius = 5
+        tableView.reloadData()
+        
+        transparentView.backgroundColor = UIColor.black.withAlphaComponent(0.9)
+        let tapGesture = UITapGestureRecognizer(target: self, action: #selector(removeTransparentView))
+        transparentView.addGestureRecognizer(tapGesture)
+        UIView.animate(withDuration: 0.4, delay: 0.0, usingSpringWithDamping: 1.0, initialSpringVelocity: 1.0, options: .curveEaseInOut, animations: {
+            self.transparentView.alpha = 0.5
+            self.tableView.frame = CGRect(x: rect.origin.x, y: rect.origin.y+rect.height, width: rect.width, height: 200)
+        }, completion: nil)
+        
+    }
+    
+    @objc func removeTransparentView() {
+        let rect = selectedButton.frame
+        UIView.animate(withDuration: 0.4, delay: 0.0, usingSpringWithDamping: 1.0, initialSpringVelocity: 1.0, options: .curveEaseInOut, animations: {
+            self.transparentView.alpha = 0
+            self.tableView.frame = CGRect(x: rect.origin.x, y: rect.origin.y+rect.height, width: rect.width, height: 0)
+        }, completion: nil)
+    }
+    
+    var leadArray = [LeadCollectionFields]()
```

Comment:
Same here: move it to top, change the name and make it private if it's not being used from outside.

Reply:
done